config.yaml

```diff
@@ -355,3 +355,13 @@ options:
       The snap channel to install the prometheus-ovs-exporter from. Setting
       this option to an empty string will result in the snap not being
       installed or removed if it has already been installed.
+  availability-zone-mapping:
```

Comment:
@rodrigogansobarbieri Do you think the config approach is superior to using the value supplied from an environment variable supplied by Juju in the hook context called `JUJU_AVAILABILITY_ZONE`?

I would rather see `JUJU_AVAILABILITY_ZONE` being used since the config option puts an extra burden on the operator whereas using the environment variable would automatically make use of the Juju-supplied metadata from the cloud provider.

Reply:
Hi Dmitrii, in othis particular case, for the purpose this feature has been asked for, I believe it is a big drawback to restrict or link the config value to the JUJU_AVAILABILITY_ZONE.

The purpose of this feature is not necessarily to match the real physical AZs, but actually to define router aggregates. Consider that you have some hosts where you have either trusted or untrusted workloads, and you also have 2 external networks for that purpose. The fact that you can create 2 OVN Router "AZs" named trusted and untrusted, you can segregate the router traffic dedicated to the associated networks to certain hosts. Otherwise, the OVN routers, being HA, will be allocated and flowing through every node that runs ovn-controller services.

Using JUJU_AVAILABILITY_ZONE will not serve that purpose, but it will serve the purpose of matching physical AZs seamlessly as you suggested.

For that trusted/untrusted use case. Indeed for 3 physical AZs scenario there would be "trusted_router_AZ1", "trusted_router_AZ2", "trusted_router_AZ3", "untrusted_router_AZ1", ... and so on. I asked about AZ names to the neutron folks upstream and they suggested that it is actually better to have different AZ names to show what that AZ configuration is related to, as the names for AZs between nova, neutron, cinder, etc, do not need to match, and it is quite beneficial that they don't, because some may have slightly different semantics.

Reply:
Having this context, I would instead suggest something different for the target use-case rather than trying to utilize availability zones for that. Availability zones in Neutron aren't meant to be a mechanism for traffic segregation - they were introduced to help Neutron decide how to place L3 or DHCP services (network namespaces implementing routers or DHCP agents in the old implementation). So it's about network service availability, not traffic policy enforcement.

If your network somehow treats different nodes differently (presumably different configs on different DC switch ports), then you are trying to simulate two distinct physical networks (or fabrics). Therefore, you can use the "physical network" concept in Neutron in order to do that.

```
juju config ovn-chassis-trusted bridge-interface-mappings='br-provider:bond-ext' ovn-bridge-mappings='dcfabric-trusted:br-provider'

juju config ovn-chassis-untrusted bridge-interface-mappings='br-provider:bond-ext' ovn-bridge-mappings='dcfabric-untrusted:br-provider'
```

In this case, you would have two external networks created with different physnets: `dcfabric-trusted` and `dcfabric-untrusted`.

The scheduling decisions will be done by Neutron according to the availability of access to physical networks (based on their physnet labels) and the connectivity of a particular router to a given external network: https://github.com/openstack/neutron/blob/320f54eba1a82917e4f02244ea8ddf9757d8f39f/neutron/plugins/ml2/drivers/ovn/mech_driver/ovsdb/ovn_client.py#L1480-L1491

The visibility of networks and their use as external networks can be controlled by cloud admins by sharing a particular physical network as external using neutron RBAC: https://docs.openstack.org/neutron/latest/admin/config-rbac.html#allowing-a-network-to-be-used-as-an-external-network

As a result, availability is not conflated with traffic segregation.